packages/edit-post/src/store/actions.js

```diff
@@ -157,53 +146,43 @@ export function removeEditorPanel( panelName ) {
  *
  * @param {string} feature Feature name.
  */
-export function* toggleFeature( feature ) {
-	yield controls.dispatch(
-		interfaceStore.name,
-		'toggleFeature',
-		'core/edit-post',
-		feature
-	);
-}
+export const toggleFeature = ( feature ) => ( { registry } ) =>
+	registry
+		.dispatch( interfaceStore )
+		.toggleFeature( 'core/edit-post', feature );
 
-export function* switchEditorMode( mode ) {
-	yield {
+export const switchEditorMode = ( mode ) => ( { dispatch, registry } ) => {
+	dispatch( {
 		type: 'SWITCH_MODE',
 		mode,
-	};
+	} );
 
 	// Unselect blocks when we switch to the code editor.
 	if ( mode !== 'visual' ) {
-		yield controls.dispatch( blockEditorStore, 'clearSelectedBlock' );
+		registry.dispatch( blockEditorStore ).clearSelectedBlock();
 	}
 
 	const message =
 		mode === 'visual'
 			? __( 'Visual editor selected' )
 			: __( 'Code editor selected' );
 	speak( message, 'assertive' );
-}
+};
 
 /**
  * Triggers an action object used to toggle a plugin name flag.
  *
  * @param {string} pluginName Plugin name.
  */
-export function* togglePinnedPluginItem( pluginName ) {
-	const isPinned = yield controls.select(
-		interfaceStore,
-		'isItemPinned',
-		'core/edit-post',
-		pluginName
-	);
+export const togglePinnedPluginItem = ( pluginName ) => ( { registry } ) => {
+	const isPinned = registry
+		.select( interfaceStore )
+		.isItemPinned( 'core/edit-post', pluginName );
 
-	yield controls.dispatch(
-		interfaceStore,
-		isPinned ? 'unpinItem' : 'pinItem',
-		'core/edit-post',
-		pluginName
-	);
-}
+	registry
+		.dispatch( interfaceStore )
+		[ isPinned ? 'unpinItem' : 'pinItem' ]( 'core/edit-post', pluginName );
```

Comment:
I had to stop here for a moment to consider all the logic. Perhaps it would read smoother without the ternary operator?

```suggestion
	const { unpinItem, pinItem} = registry.dispatch( interfaceStore );
	if ( isPinned ) {
		unpinItem( 'core/edit-post', pluginName );
	} else {
		pinItem( 'core/edit-post', pluginName );
	}
```

Reply:
Ideally (out of scope for this PR) there would be a `togglePinnedItem` action in the `interface` store. The `pinItem`/`unpinItem` actions are used at two places, and both of them really want to toggle.